File: Project Academy/07. LiveData-test/java/app/src/main/java/com/dicoding/academies/ui/reader/CourseReaderViewModel.java
package com.dicoding.academies.ui.reader;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.dicoding.academies.data.AcademyRepository;
import com.dicoding.academies.data.source.local.entity.ModuleEntity;

import java.util.List;


public class CourseReaderViewModel extends ViewModel {

    private String courseId;
    private String moduleId;
    private AcademyRepository academyRepository;

    public CourseReaderViewModel(AcademyRepository mAcademyRepository) {
        this.academyRepository = mAcademyRepository;
    }

    public void setSelectedCourse(String courseId) {
        this.courseId = courseId;
    }

    public void setSelectedModule(String moduleId) {
        this.moduleId = moduleId;
    }

    public LiveData<List<ModuleEntity>> getModules() {
        return academyRepository.getAllModulesByCourse(courseId);
    }

    public LiveData<ModuleEntity> getSelectedModule() {
        return academyRepository.getContent(courseId, moduleId);
    }

}

File: Project Academy/6. LiveData/kotlin/app/src/main/java/com/dicoding/academies/ui/reader/CourseReaderViewModel.kt
package com.dicoding.academies.ui.reader

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.dicoding.academies.data.AcademyRepository
import com.dicoding.academies.data.source.local.entity.ModuleEntity

class CourseReaderViewModel(private val academyRepository: AcademyRepository) : ViewModel() {

    var courseId: String? = null
    var moduleId: String? = null

    fun getModules(): LiveData<List<ModuleEntity>> {
        return academyRepository.getAllModulesByCourse(courseId)
    }

    fun getSelectedModule(): LiveData<ModuleEntity> {
        return academyRepository.getContent(courseId, moduleId)
    }
}

